app/controllers/overrides/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,51 @@
+module Overrides
+  class OmniauthCallbacksController <
+    DeviseTokenAuth::OmniauthCallbacksController
+    def assign_provider_attrs(user, auth_hash)
+      all_attrs = auth_hash['info'].slice(*user.attributes.keys)
+      orig_val = ActionController::Parameters.permit_all_parameters
+      ActionController::Parameters.permit_all_parameters = true
+      permitted_attrs = ActionController::Parameters.new(all_attrs)
+      permitted_attrs.permit({})
+      user.assign_attributes(permitted_attrs)
+      ActionController::Parameters.permit_all_parameters = orig_val
+      user
+    end
+
+    def get_resource_from_auth_hash
+      logger.debug auth_hash
+      # find or create user by provider and provider uid
```

Comment:
It's no longer find or create

Reply:
Are you sure we can no longer use find or create or some variant?

Reply:
To use find_or_create, we would need to provide all of the required attributes to the method. But in our case we may only provide the provider and uid, in case the user has changed any of the other fields in their Facebook profile.

Reply:
Actually, having thought about this, we need to update the user's data even if they already exist, in case they've changed their Facebook profile info.

Reply:
I have removed the overriden method since I figured out it was redundant and the original one is already doing what we want.